operator/controllers/seldondeployment_controller_test.go

```diff
@@ -711,6 +711,118 @@ var _ = Describe("Create a Seldon Deployment and then a new one", func() {
 
 })
 
+var _ = Describe("Create a Seldon Deployment with long name", func() {
+	const timeout = time.Second * 30
+	const interval = time.Second * 1
+	namespaceName := rand.String(10)
+	name := "seldon-model-1234567890-1234567890-1234567890-1234567890-1234567890"
+	By("Creating a resource")
+	It("should create a resource with defaults", func() {
+		Expect(k8sClient).NotTo(BeNil())
+		var modelType = machinelearningv1.MODEL
+		key := types.NamespacedName{
+			Name:      name,
+			Namespace: namespaceName,
+		}
+		instance := &machinelearningv1.SeldonDeployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      key.Name,
+				Namespace: key.Namespace,
+			},
+			Spec: machinelearningv1.SeldonDeploymentSpec{
+				Name: "mydep",
+				Predictors: []machinelearningv1.PredictorSpec{
+					{
+						Name: "p1",
+						ComponentSpecs: []*machinelearningv1.SeldonPodSpec{
+							{
+								Spec: v1.PodSpec{
+									Containers: []v1.Container{
+										{
+											Image: "seldonio/mock_classifier:1.0",
+											Name:  "classifier",
+										},
+									},
+								},
+							},
+						},
+						Graph: &machinelearningv1.PredictiveUnit{
+							Name: "classifier",
+							Type: &modelType,
+						},
+					},
+				},
+			},
+		}
+
+		//Create namespace
+		namespace := &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		}
+		Expect(k8sClient.Create(context.Background(), namespace)).Should(Succeed())
```

Comment:
We could probably move most of this setup (e.g. creating the namespace, defining a `SeldonDeployment` spec, setting default timeouts, etc.) into a `BeforeEach` block shared by all the tests in `operator/controllers/seldondeployment_controller_test.go`.

This is just a thought though and it probably falls outside the scope of this PR, so feel free to ignore.